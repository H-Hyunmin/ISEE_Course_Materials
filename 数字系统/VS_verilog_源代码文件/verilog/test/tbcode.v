//~ `New testbench




`timescale 1ns/1ps
`include "testcode.v"






module tb_half_adder;

// half_adder Parameters
parameter PERIOD  = 10;


// half_adder Inputs
reg   a                                    = 0 ;
reg   b                                    = 0 ;

// half_adder Outputs
wire  sum                                  ;
wire  cout                                 ;




half_adder  u_half_adder (
    .a                       ( a      ),
    .b                       ( b      ),

    .sum                     ( sum    ),
    .cout                    ( cout   )
);
parameter clk_period = 10;  
reg clk;  
initial  
    clk = 0;  
always #(clk_period/2) clk = ~clk;  
always @(negedge clk) begin
    a=$random%2;
    b=$random%2;
end

initial
begin
$dumpfile("testcode.vcd");
$dumpvars(0,tb_half_adder);
    #10000
    $finish;
end



endmodule