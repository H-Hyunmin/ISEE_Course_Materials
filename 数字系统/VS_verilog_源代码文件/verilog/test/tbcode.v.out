#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1110-g18392a46)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_0000023bb840b250 .scope module, "tb_half_adder" "tb_half_adder" 2 14;
 .timescale -9 -12;
P_0000023bb84232f0 .param/l "PERIOD" 0 2 17, +C4<00000000000000000000000000001010>;
P_0000023bb8423328 .param/l "clk_period" 0 2 38, +C4<00000000000000000000000000001010>;
v0000023bb8422de0_0 .var "a", 0 0;
v0000023bb8422e80_0 .var "b", 0 0;
v0000023bb8422f20_0 .var "clk", 0 0;
v0000023bb8422fc0_0 .net "cout", 0 0, L_0000023bb84087c0;  1 drivers
v0000023bb8423060_0 .net "sum", 0 0, L_0000023bb8423100;  1 drivers
E_0000023bb8446d10 .event negedge, v0000023bb8422f20_0;
S_0000023bb840b970 .scope module, "u_half_adder" "half_adder" 2 31, 3 1 0, S_0000023bb840b250;
 .timescale -9 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000023bb8423100 .functor XOR 1, v0000023bb8422de0_0, v0000023bb8422e80_0, C4<0>, C4<0>;
L_0000023bb84087c0 .functor AND 1, v0000023bb8422de0_0, v0000023bb8422e80_0, C4<1>, C4<1>;
v0000023bb8408720_0 .net "a", 0 0, v0000023bb8422de0_0;  1 drivers
v0000023bb8456220_0 .net "b", 0 0, v0000023bb8422e80_0;  1 drivers
v0000023bb8422ca0_0 .net "cout", 0 0, L_0000023bb84087c0;  alias, 1 drivers
v0000023bb8422d40_0 .net "sum", 0 0, L_0000023bb8423100;  alias, 1 drivers
    .scope S_0000023bb840b250;
T_0 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000023bb8422de0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000023bb8422e80_0, 0, 1;
    %end;
    .thread T_0;
    .scope S_0000023bb840b250;
T_1 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000023bb8422f20_0, 0, 1;
    %end;
    .thread T_1;
    .scope S_0000023bb840b250;
T_2 ;
    %delay 5000, 0;
    %load/vec4 v0000023bb8422f20_0;
    %inv;
    %store/vec4 v0000023bb8422f20_0, 0, 1;
    %jmp T_2;
    .thread T_2;
    .scope S_0000023bb840b250;
T_3 ;
    %wait E_0000023bb8446d10;
    %vpi_func 2 44 "$random" 32 {0 0 0};
    %pushi/vec4 2, 0, 32;
    %mod/s;
    %pad/s 1;
    %store/vec4 v0000023bb8422de0_0, 0, 1;
    %vpi_func 2 45 "$random" 32 {0 0 0};
    %pushi/vec4 2, 0, 32;
    %mod/s;
    %pad/s 1;
    %store/vec4 v0000023bb8422e80_0, 0, 1;
    %jmp T_3;
    .thread T_3;
    .scope S_0000023bb840b250;
T_4 ;
    %vpi_call 2 50 "$dumpfile", "testcode.vcd" {0 0 0};
    %vpi_call 2 51 "$dumpvars", 32'sb00000000000000000000000000000000, S_0000023bb840b250 {0 0 0};
    %delay 10000000, 0;
    %vpi_call 2 53 "$finish" {0 0 0};
    %end;
    .thread T_4;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "tbcode.v";
    "./testcode.v";
